rts/at/annotation/common-service/src/main/java/io/narayana/sra/demo/model/Booking.java

```diff
@@ -0,0 +1,252 @@
+
+        return Arrays.asList(mapper.readValue(json, Booking[].class));
+    }
+
+    private void init(String id, String name, Integer quantity, String type, BookingStatus status, Booking[] details) {
+        String[] segments = id.split("/");
+
+        this.id = segments[segments.length - 1]; // this is just a demo so don't check for a single /
+        this.name = name == null ? "" : name;
+        this.quantity = quantity;
+        this.type = type == null ? "" : type;
+        this.status = status;
+        this.details = details == null ? new Booking[0] : removeNullEnElements(details);
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> T[] removeNullEnElements(T[] a) {
+        List<T> list = new ArrayList<T>(Arrays.asList(a));
+        list.removeAll(Collections.singleton(null));
+        return list.toArray((T[]) Array.newInstance(a.getClass().getComponentType(), list.size()));
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Integer getQuantity() {
+        return quantity;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public Booking[] getDetails() {
+        return details;
+    }
+
+    public BookingStatus getStatus() {
+        return status;
+    }
+
+    public void setStatus(BookingStatus status) {
+        this.status = status;
+    }
+
+    public void requestCancel() {
+        status = BookingStatus.CANCEL_REQUESTED;
+    }
+
+    public void setConfirmed() {
+        status = BookingStatus.CONFIRMED;
+    }
+
+    public String toString() {
+        return String.format("{\"id\":\"%s\",\"name\":\"%s\",\"quantity\":\"%d\",\"type\":\"%s\",\"status\":\"%s\"}",
+                id, name, quantity, type, status);
+    }
+
+    public void setCanceled() {
+        status = BookingStatus.CANCELLED;
+    }
+
+    public void setConfirming() {
+        status = BookingStatus.CONFIRMING;
+    }
+
+    @JsonIgnore
+    public boolean isCancelPending() {
+        return status == BookingStatus.CANCEL_REQUESTED;
+    }
+
+    public boolean merge(Booking booking) {
+        if (!id.equals(booking.getId()))
+            return false; // or throw an exception
+
+        name = booking.getName();
+        quantity = booking.getQuantity();
+        status = booking.getStatus();
+
+        return true;
+    }
+
+    @JsonIgnore
+    public String getEncodedId() {
+        try {
+            return URLEncoder.encode(id, "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            return id; // TODD do it in the constructor
+        }
+    }
+
+    public String toJson() throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        return objectMapper.writeValueAsString(this);
+    }
+
+    @JsonIgnore
+    public IOException getDecodingException() {
+        return decodingException;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Why the re-implementation of `equals`, what problem is if fixing?

The booking id is a business concept, if you look through the demonstrator you will find how the booking id is used. There are a few things to say about your "new definition" of the  booking id but I will say that as you have re-coded it, as the booking transitions through its state model it's id will morph.

Reply:
> Why the re-implementation of `equals`, what problem is it fixing?
> 
> The booking id is a business concept, if you look through the demonstrator you will find how the booking id is used. There are a few things to say about your "new definition" of the booking id but I will say that as you have re-coded it, as the booking transitions through its state model it's id will morph.

Hi @mmusgrov I've not re-implemented equals method in Booking, it has come as it is from narayana along with history.

Reply:
I agree with @moulalis . It is hard to see it from  'PR view'. But you can check it better from 'blame view' (https://github.com/moulalis/quickstart-1/blame/JBTM-3462/rts/at/annotation/common-service/src/main/java/io/narayana/sra/demo/model/Booking.java#L209-L222)

Reply:
Hi @mmusgrov , Rather then "Migrate SRA demo" can we keep other part as a separate issue/discussion?  Putting everything in the same PR is however creating a mess.

Reply:
I didn't understand your last message.

@marcosgopen I know you said you had tested an earlier version of this PR but do you know if it still works?

Reply:
@moulalis can you add copyrights to the files please.

Reply:
There is no run script or automatic testing of the quickstart so we won't be able to test it on CI.

The instructions for starting the services all try to start on debugger port 5005 so the second services fails to start. When I changed each service to use a different debugger port it fails with `[mmusgrov@dev2 annotation] ((HEAD detached at pr/434)) $ curl -XPOST http://localhost:8082/trip/book?hotelName=Rex
curl: (7) Failed to connect to localhost port 8082 after 0 ms: Connection refused`

If we can have an automated way of testing this in CI then it might be easier for me to figure out how to run the example.

Reply:
Hi @mmusgrov copy rights added to the required file. I've also configured a pom.xml of each service that can run on different debug port. Please try it now. Thanks.

Reply:
> I didn't understand your last message.
What you didn't understand? is this question is for me or @marcosgopen ?

Reply:
> I didn't understand your last message.
> 
> @marcosgopen I know you said you had tested an earlier version of this PR but do you know if it still works?

@mmusgrov yes, I tested it again and it worked. I remember I saw the debugger port issue, but it does not prevent the demo from working. Thanks @moulalis for fixing that.

Reply:
> Hi @mmusgrov , Rather then "Migrate SRA demo" can we keep other part as a separate issue/discussion? Putting everything in the same PR is however creating a mess.

I think I understand what you mean @moulalis and I might answer. We need to fix what Mike suggested ('equals' method issue, see above) if we want to have this demo in quickstart. So I agree we need to have it fixed it in this PR, with a separate commit.
Generally speaking when we know there is a problem in the code (Mike found it) we want to have it fixed before adding it to the repo.

Reply:
> 
> If we can have an automated way of testing this in CI then it might be easier for me to figure out how to run the example.

+1

Reply:
> @moulalis can you add copyrights to the files please.

@mmusgrov IIUC we want to remove copyright and licence from quickstart's files (see PR https://github.com/jbosstm/quickstart/pull/438) as written here https://github.com/jbosstm/quickstart/blob/main/CONTRIBUTING.md?plain=1#L13

Reply:
Yes that's correct, all contributions should follow our CONTRIBUTING.md guidelines.

Reply:
> > I didn't understand your last message.
> > Hi @mmusgrov What you didn't understand? is this question is for me or @marcosgopen ?

The [message immediately before the one where indicated that I didn't understand](https://github.com/jbosstm/quickstart/pull/434#discussion_r1254344152). If I refer to a comment that is out of sequence then I will always quote it - like I have in this comment.

Reply:
Hi @mmusgrov , plz resolve the conversation id it met the expectation. Thanks.

Reply:
> There is no run script or automatic testing of the quickstart so we won't be able to test it on CI.
> 
> The instructions for starting the services all try to start on debugger port 5005 so the second services fails to start. When I changed each service to use a different debugger port it fails with `[mmusgrov@dev2 annotation] ((HEAD detached at pr/434)) $ curl -XPOST http://localhost:8082/trip/book?hotelName=Rex curl: (7) Failed to connect to localhost port 8082 after 0 ms: Connection refused`
> 
> If we can have an automated way of testing this in CI then it might be easier for me to figure out how to run the example.

I don't recall seeing your response to this review request.

Reply:
> > There is no run script or automatic testing of the quickstart so we won't be able to test it on CI.
> > The instructions for starting the services all try to start on debugger port 5005 so the second services fails to start. When I changed each service to use a different debugger port it fails with `[mmusgrov@dev2 annotation] ((HEAD detached at pr/434)) $ curl -XPOST http://localhost:8082/trip/book?hotelName=Rex curl: (7) Failed to connect to localhost port 8082 after 0 ms: Connection refused`
> > If we can have an automated way of testing this in CI then it might be easier for me to figure out how to run the example.
> 
> I don't recall seeing your response to this review request.

Hi @mmusgrov I've changed each service to use different debug port in pom.xml of all the, so now you can start the server and can try to connect debugger based on the port.  regarding automated way of testing this in CI I need to look for it if it is really required, I can look into it once every thing is finished.